aviatrix/data_source_aviatrix_gateway.go

```diff
@@ -505,19 +505,19 @@ func dataSourceAviatrixGatewayRead(d *schema.ResourceData, meta interface{}) err
 			d.Set("peering_ha_public_ip", gwHaGw.PublicIP)
 			d.Set("peering_ha_gw_size", gwHaGw.GwSize)
 			d.Set("peering_ha_private_ip", gwHaGw.PrivateIP)
-			if gwHaGw.CloudType == goaviatrix.AWS || gwHaGw.CloudType == goaviatrix.AWSGOV {
+			if goaviatrix.IsCloudType(gwHaGw.CloudType, goaviatrix.AWSRelatedCloudTypes) {
 				d.Set("peering_ha_subnet", gwHaGw.VpcNet)
 				if gwHaGw.InsaneMode == "yes" {
 					d.Set("peering_ha_insane_mode_az", gwHaGw.GatewayZone)
 				}
-			} else if gwHaGw.CloudType == goaviatrix.AZURE || gwHaGw.CloudType == goaviatrix.OCI {
+			} else if goaviatrix.IsCloudType(gwHaGw.CloudType, goaviatrix.AZURECHINA|goaviatrix.OCIRelatedCloudTypes) {
 				d.Set("peering_ha_subnet", gwHaGw.VpcNet)
-			} else if gwHaGw.CloudType == goaviatrix.GCP {
+			} else if goaviatrix.IsCloudType(gwHaGw.CloudType, goaviatrix.GCPRelatedCloudTypes) {
 				d.Set("peering_ha_zone", gwHaGw.GatewayZone)
 			}
```

Comment:
What happens if cloud_type == AZURE? I don't see a branch for AZURE anymore

Reply:
It should be all azure clouds not just azure china.